lib/createNavLink.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright 2015, Yahoo! Inc.
+ * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+ */
+/*global window */
+'use strict';
+var React = require('react');
+var RouteStore = require('./RouteStore');
+var debug = require('debug')('NavLink');
+var navigateAction = require('./navigateAction');
+var objectAssign = require('object-assign');
+
+function isLeftClickEvent (e) {
+    return e.button === 0;
+}
+
+function isModifiedEvent (e) {
+    return !!(e.metaKey || e.altKey || e.ctrlKey || e.shiftKey);
+}
+
+/**
+ * create NavLink with custom options
+ * @param {Object} options to overwrite the default attributes to create NavLink
+ * @returns {React.Component} NavLink
+ */
+module.exports = function createNavLink (options) {
+    var NavLink = React.createClass(objectAssign({}, {
+        displayName: 'NavLink',
+        contextTypes: {
+            executeAction: React.PropTypes.func,
+            getStore: React.PropTypes.func
+        },
+        propTypes: {
+            currentRoute: React.PropTypes.object,
+            currentNavigate: React.PropTypes.object,
```

Comment:
Who will populate currentRoute and currentNavigate props, after `handleRoute` is removed?